website/docs/docs/contributing/testing-a-new-adapter.md

```diff
@@ -0,0 +1,200 @@
+
+3. **Custom tests**, for behavior that is specific to your adapter / data platform. Each data warehouse has its own specialties and idiosyncracies. We encourage you to use the same `pytest`-based framework, utilities, and fixtures to write your own custom tests for functionality that is unique to your adapter.
+
+If you run into an issue with the core framework, or the basic/optional test cases—or if you've written a custom test that you believe would be relevant and useful for other adapter plugin developers—please open an issue or PR in the `dbt-core` repository on GitHub.
+
+## Steps to get started
+
+1. Install dependencies
+2. Set up and configure pytest
+3. Define test cases
+
+### Install dependencies
+
+You should already have a virtual environment with `dbt-core` and your adapter plugin installed. You'll also need to install:
+- `pytest`
+- `dbt-tests-adapter`, the set of basic test cases
+
+During initial development, the `dbt-tests-adapter` package is not yet on PyPi, so please install it directly from GitHub:
+```bash
+pip install "git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-tests-adapter&subdirectory=tests/adapter"
+```
+
+Or specify it in a requirements file like:
+<File name="dev_requirements.txt">
+
+```txt
+git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-tests-adapter&subdirectory=tests/adapter
+```
+</File>
+
+### Set up and configure pytest
+
+First, set yourself up to run `pytest` by creating a file named `pytest.ini` at the root of your repository:
+
+<File name="pytest.ini">
+
+```python
+[pytest]
+filterwarnings =
+    ignore:.*'soft_unicode' has been renamed to 'soft_str'*:DeprecationWarning
+    ignore:unclosed file .*:ResourceWarning
+env_files =
+    test.env  # this allows you to use env vars from a file named test.env
+              # be sure to add "test.env" to .gitignore as well!
+testpaths =
+    tests/functional  # name per convention
+```
+
+</File>
+
+Then, create a configuration file within your tests directory. In it, you'll want to define all necessary profile configuration for connecting to your data platform in local development and continuous integration. We recommend setting these values with environment variables, since this file will be checked into version control.
+
+<File name="pytest.ini">
+
+```python
+import pytest
+import os
+
+# Import the standard functional fixtures as a plugin
+# Note: fixtures with session scope need to be local
+pytest_plugins = ["dbt.tests.fixtures.project"]
+
+# The profile dictionary, used to write out profiles.yml
+@pytest.fixture(scope="class")
+def dbt_profile_target(unique_schema):
+    return {
+        'type': '<myadapter>',
+        'threads': 1,
+        'host': os.getenv('HOST_ENV_VAR_NAME'),
+        'user': os.getenv('USER_ENV_VAR_NAME'),
+        ...
+        'schema': unique_schema  # this will be passed in by the testing framework
```

Comment:
In the Redshift test I specified the unique_schema, but in at least some of the other adapters I didn't. I'm also setting it in the 'dbt_profile_data' fixture, so it's not actually necessary. I'm thinking we should standardize on not supplying it here.